make/libraries

```diff
@@ -49,48 +51,72 @@ $(SUNDIALS)/lib/libsundials_nvecserial.a: $(SUNDIALS_NVECSERIAL)
 	@mkdir -p $(dir $@)
 	$(AR) -rs $@ $^
 
-LIBSUNDIALS = $(SUNDIALS)/lib/libsundials_nvecserial.a $(SUNDIALS)/lib/libsundials_cvodes.a $(SUNDIALS)/lib/libsundials_idas.a
+LIBSUNDIALS := $(SUNDIALS)/lib/libsundials_nvecserial.a $(SUNDIALS)/lib/libsundials_cvodes.a $(SUNDIALS)/lib/libsundials_idas.a
 
-STAN_CVODES_HEADERS := $(shell find stan -name *cvodes*.hpp)
-$(STAN_CVODES_HEADERS) : $(LIBSUNDIALS)
+STAN_SUNDIALS_HEADERS := $(shell find $(MATH)stan -name *cvodes*.hpp) $(shell find $(MATH)stan -name *idas*.hpp)
+$(STAN_SUNDIALS_HEADERS) : $(LIBSUNDIALS)
 
-STAN_IDAS_HEADERS := $(shell find stan -name *idas*.hpp)
-$(STAN_IDAS_HEADERS) : $(LIBSUNDIALS)
+clean-sundials:
+	$(RM) $(wildcard $(sort $(SUNDIALS_CVODES) $(SUNDIALS_IDAS) $(SUNDIALS_NVECSERIAL) $(LIBSUNDIALS)))
+
+
+############################################################
+# MPI build rules
+
+ifdef STAN_MPI
 
 ##
-# Build rules for MPI
-##
+# Boost build options
+BOOST_PARALLEL_JOBS ?= 1
 
-# turn BOOST_LIB into absolute path
-BOOST_LIB_ABS = $(shell mkdir -p "$(BOOST_LIB)" && cd -- "$(BOOST_LIB)" && pwd)
 
 $(BOOST)/user-config.jam:
 	echo "# In case of a compiler mismatch used by mpicxx and" >> $(BOOST)/user-config.jam
 	echo "# the compiler used for Stan, consider configuring" >> $(BOOST)/user-config.jam
 	echo "# the boost toolset here" >> $(BOOST)/user-config.jam
```

Comment:
What turns out to be very useful is to specifically direct boost to a specific mpicxx with this statement:

`using mpi : /path/to/mpicxx ;`

How about we include a comment here which mentions this possibility? This is not necessary, but it is helpful on systems with multiple MPI installations.